src/shared/resolve_or_timeout.test.ts

```diff
@@ -0,0 +1,33 @@
+/*function timeout(millis: number): Promise<void> {
+    return new Promise((resolve) => {
+      setTimeout(() => {
+        resolve();
+      }, millis);
+    });
+  }*/
+
+import { resolveOrTimeout } from "./resolve_or_timeout";
+
+test("Case where promise doesn't timeout", async () => {
+  const testPromise = new Promise<string>((resolve) => {
+    setTimeout(() => {
+      resolve("Promise completed");
+    }, 100);
+  });
+  const temp = await resolveOrTimeout(testPromise, 200);
+  expect(temp.result).toBe("Promise completed");
+  expect(temp.timedOut).toBe(false);
+  expect(temp.promise).toBeUndefined();
+});
+
+test("Case where promies DOES timeout", async () => {
+  const testPromise = new Promise<string>((resolve) => {
```

Comment:
timeout helper from utils

Reply:
Also no need for a temp var.
`const result = await resolveOrTimeout(timeout(200), 100)`

Reply:
You could put this back in a temp var to do result.promise validation, but I'd instead recommend checking the return value of the promise. Something like
`
const result = await resolveOrTimeout((async () => {
  await timeout(200);
  return "test value";
})(), 100);
...
expect(await result.promise).toBe("test value");

Reply:
Instead, I changed toBe() to become toStrictEquals() to handle the issue. Test now passes.

Reply:
Change was reverted when the kismarton machine broke down, I have once again made this change.